app/uk/gov/hmrc/agentaccesscontrol/connectors/GovernmentGatewayProxyConnector.scala

```diff
@@ -39,15 +40,17 @@ case class AssignedAgent(allocatedAgentCode: AgentCode) {
 
 }
 
-class GovernmentGatewayProxyConnector(baseUrl: URL, httpPost: HttpPost, auditService: AuditService) {
+class GovernmentGatewayProxyConnector(baseUrl: URL, httpPost: HttpPost, auditService: AuditService) extends HttpAPIMonitor {
   val url: URL = new URL(baseUrl, "/government-gateway-proxy/api/admin/GsoAdminGetAssignedAgents")
 
   def getAssignedSaAgents(utr: SaUtr, agentCode: AgentCode)(implicit hc: HeaderCarrier): Future[Seq[AssignedAgent]] = {
-    httpPost.POSTString(url.toString, body(utr), Seq(CONTENT_TYPE -> XML))
+    monitor("ConsumedAPI-GGW-GetAssignedAgents-POST"){
+      httpPost.POSTString(url.toString, body(utr), Seq(CONTENT_TYPE -> XML))
```

Comment:
It is; however, WSHTTP::POSTString is not overridden. Changes would be necessary in the monitoring library.